File: FunctionalBlazor/FunctionalBlazor.Composition/Startup.fs
namespace FunctionalBlazor.Composition

open Microsoft.Extensions.DependencyInjection
open System.Runtime.CompilerServices
open FunctionalBlazor.Common.Ioc
open FunctionalBlazor.Composition

[<Extension>]
type Startup  () =
        
    [<Extension>]
    static member inline ConfigureFunctions (services : IServiceCollection) =
        services.AddScoped<IFunctionRoot, FunctionRoot>() |> ignore

File: FunctionalBlazor/FunctionalAndroid/Startup.cs
using static FunctionalBlazor.Program.Startup;
using static FunctionalBlazor.Caching.Startup;
using static FunctionalBlazor.Composition.Startup;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using FunctionalXamarin;

namespace FunctionalAndroid
{
    public class Startup
    {
        internal static void ConfigureServices(HostBuilderContext ctx, IServiceCollection services)
        {
            services.ConfigureCaching();
            services.ConfigureProgram();
            services.ConfigureViewModels();
            services.ConfigureFunctions();
        }
    }
}